### PR TITLE
feat: [PL-19818]: Added CustomSecretManager connector behind FF

### PR DESCRIPTION
**Pull request by meenakshi-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>
Screenshot 2022-08-18 at 11 15 23 AM
Screenshot 2022-08-18 at 11 15 35 AM
Screenshot 2022-08-18 at 11 17 07 AM
Screenshot 2022-08-18 at 11 17 15 AM
Screenshot 2022-08-18 at 11 17 32 AM

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 72 Issues 
* Code issue 72 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10467)

**Comment by meenakshi-harness** [comment]
retrigger coverage

**Comment by meenakshi-harness** [comment]
retrigger coverage

**Comment by meenakshi-harness** [comment]
retrigger coverage

**Comment by meenakshi-harness** [comment]
retrigger titlecheck

File: src/modules/10-common/components/ScriptVariableRuntimeInput/ScriptVariablesRuntimeInput.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import { isArray, isEmpty } from 'lodash-es'
+import { AllowedTypes, FormInput, SelectOption } from '@harness/uicore'
+import cx from 'classnames'
+import { useStrings } from 'framework/strings'
+import MultiTypeFieldSelector from '@common/components/MultiTypeFieldSelector/MultiTypeFieldSelector'
+import css from './ScriptVariablesRuntimeInput.module.scss'
+
+interface InputOutputVariablesInputSetProps {
+  allowableTypes: AllowedTypes
+  // todo: to change with type in BE
```
Comment (pbarapatre10) [inline src/modules/10-common/components/ScriptVariableRuntimeInput/ScriptVariablesRuntimeInput.tsx:19]: is the BE change merged? can remove this, in that case

Comment (meenakshi-harness) [reply src/modules/10-common/components/ScriptVariableRuntimeInput/ScriptVariablesRuntimeInput.tsx:19]: Not yet.. will update once merged

File: src/modules/10-common/components/ScriptVariableRuntimeInput/ScriptVariablesRuntimeInput.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+                            disabled={true}
+                          />
+                          <FormInput.MultiTextInput
+                            name={`${prefix}templateInputs.environmentVariables[${i}].value`}
+                            multiTextInputProps={{
+                              allowableTypes,
+
+                              disabled: readonly
+                            }}
+                            label=""
```
Comment (pbarapatre10) [inline src/modules/10-common/components/ScriptVariableRuntimeInput/ScriptVariablesRuntimeInput.tsx:91]: label is supposed to be empty? can we remove it.

Comment (meenakshi-harness) [reply src/modules/10-common/components/ScriptVariableRuntimeInput/ScriptVariablesRuntimeInput.tsx:91]: This is a required filed.. we don't need label hence passed empty string

**Review by pbarapatre10 — APPROVED** [review]
LGTM

**Review by upadhyay-an — DISMISSED** [review]

**Review by upadhyay-an — APPROVED** [review]

**Comment by meenakshi-harness** [comment]
retrigger jest

**Comment by meenakshi-harness** [comment]
retrigger coverage

**Pull request closed without merging**